apis/python/src/tiledbsoma/io/ingest.py

```diff
@@ -510,10 +510,12 @@ def from_anndata(
                 #   chunkwise into memory.
                 # Using the latter allows us to ingest larger .h5ad files without OOMing.
 
-                # Some AnnData objects have no X at all.
+                # Some AnnData objects have no X at all. This might be a missing attribute
+                # anndata.X; it might be anndata.X present, but with value None.
                 has_X = True
                 try:
                     anndata.X
+                    has_X = anndata.X is not None
```

Comment:
I would just say `has_X = anndata.X is not None` and remove the `anndata.X` and `has_X = True` above, like:

```python
# Some AnnData objects ...
try:
    has_X = anndata.X is not None
except ...:
    ...
```

Alternately, you might restructure this to eliminate the `has_X` variable entirely:

```python
try:
    anndata_X = anndata.X
except (...):
    pass
else:
    if anndata_X is not None:
        ...
```